astroquery/linelists/cdms/tests/test_cdms.py

```diff
@@ -57,9 +61,54 @@ def test_query():
     tbl = CDMS._parse_result(response)
     assert isinstance(tbl, Table)
     assert len(tbl) == 8
-    assert set(tbl.keys()) == set(['FREQ', 'ERR', 'LGINT', 'DR', 'ELO', 'GUP',
-                                   'TAG', 'QNFMT', 'Ju', 'Jl', "vu", "vl", "Ku", "Kl", "F", "name"])
+    assert set(tbl.keys()) == colname_set
 
     assert tbl['FREQ'][0] == 115271.2018
     assert tbl['ERR'][0] == .0005
     assert tbl['LGINT'][0] == -7.1425
+
+
+def test_parseletternumber():
+    """
+    Very Important:
+    Exactly two characters are available for each quantum number. Therefore, half
+    integer quanta are rounded up ! In addition, capital letters are used to
+    indicate quantum numbers larger than 99. E. g. A0 is 100, Z9 is 359. Small
+    types are used to signal corresponding negative quantum numbers.
+    """
+
+    # examples from the docs
+    assert parse_letternumber("A0") == 100
+    assert parse_letternumber("Z9") == 359
+
+    # inferred?
+    assert parse_letternumber("z9") == -359
+    assert parse_letternumber("ZZ") == 3535
+
+
+def test_hc7s():
+    """
+    Test for a very complicated molecule
+
+    CDMS.query_lines_async(100*u.GHz, 100.755608*u.GHz, molecule='HC7S', parse_name_locally=True)
+    """
+
+    response = MockResponseSpec('HC7S.data')
+    tbl = CDMS._parse_result(response)
```

Comment:
are you ever supposed to call `_parse_result` directly? If not the test should be refactored.

Reply:
no.  I see your point.  I was copying from an old/previous test, but it would be better to test the user-facing API here, I agree.

Reply:
ahh, yes. Then the previous test was failing, too, I suppose 😅